lib/src/SaiInterface.cpp

```diff
@@ -173,3 +173,275 @@ sai_status_t SaiInterface::get(
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t *attr_count,
+        _In_ const sai_attribute_t **attr_list,
+        _In_ sai_bulk_op_error_mode_t mode,
+        _Out_ sai_status_t *object_statuses)
+{
+    SWSS_LOG_ENTER();
+
+    sai_status_t status = SAI_STATUS_SUCCESS;
+
+    uint32_t object_count = (uint32_t)metaKey.size();
+
+    if (!object_count)
+    {
+        SWSS_LOG_ERROR("Container with objects is empty in bulkCreate");
+        return SAI_STATUS_FAILURE;
+    }
+
+    auto info = sai_metadata_get_object_type_info(metaKey[0].objecttype);
+
+    for (uint32_t idx = 0; idx < object_count; idx++)
+    {
+        if (metaKey[idx].objecttype != info->objecttype)
+        {
+            SWSS_LOG_ERROR(
+                "invalid type %s in metaKeys container with types %s",
+                metaKey[idx].objecttype,
+                info->objecttype);
+            return SAI_STATUS_FAILURE;
+        }
+    }
+
+    if (info->isobjectid)
+    {
+        std::vector<sai_object_id_t> objectIds(object_count);
+
+        status = bulkCreate(
+                    info->objecttype,
+                    switch_id,
+                    object_count,
+                    attr_count,
+                    attr_list,
+                    mode,
+                    objectIds.data(),
+                    object_statuses);
+
+        for (uint32_t idx = 0; idx < object_count; idx++)
+        {
+            metaKey[idx].objectkey.key.object_id = objectIds[idx];
+        }
+
+        return status;
+    }
+
+    switch (info->objecttype)
+    {
+        case SAI_OBJECT_TYPE_FDB_ENTRY:
+        {
+            std::vector<sai_fdb_entry_t> entries;
+            for (size_t idx = 0; idx < object_count; idx++)
+            {
+                entries.push_back(metaKey[idx].objectkey.key.fdb_entry);
+            }
+            return bulkCreate(object_count,entries.data(), attr_count, attr_list, mode, object_statuses);
+        }
+
+        case SAI_OBJECT_TYPE_ROUTE_ENTRY:
+        {
+            std::vector<sai_route_entry_t> entries;
+            for (size_t idx = 0; idx < object_count; idx++)
+            {
+                entries.push_back(metaKey[idx].objectkey.key.route_entry);
+            }
+            return bulkCreate(object_count, entries.data(), attr_count, attr_list, mode, object_statuses);
+        }
+
+        case SAI_OBJECT_TYPE_NAT_ENTRY:
+        {
+            std::vector<sai_nat_entry_t> entries;
+            for (size_t idx = 0; idx < object_count; idx++)
+            {
+                entries.push_back(metaKey[idx].objectkey.key.nat_entry);
+            }
+            return bulkCreate(object_count, entries.data(), attr_count, attr_list, mode, object_statuses);
+        }
+
+        default:
+            SWSS_LOG_ERROR("object type %s not implemented, FIXME", info->objecttypename);
+            return SAI_STATUS_NOT_SUPPORTED;
+    }
+}
+
+sai_status_t SaiInterface::bulkRemove(
```

Comment:
im also interested, since you added create/remove/set bulk APIS that handle meta_key, but we already added
170             SAIREDIS_SAIINTERFACE_DECLARE_BULK_CREATE_ENTRY(fdb_entry);
171             SAIREDIS_SAIINTERFACE_DECLARE_BULK_CREATE_ENTRY(nat_entry);
172             SAIREDIS_SAIINTERFACE_DECLARE_BULK_CREATE_ENTRY(route_entry);

and remove/set also for *_entry objects, so actually using vector of meta_key will require copy to new array of objects, like you are for example doing like in line 322, we can keep that, but from performance reason when we doing "deserialize" we already know what object type we are dealing with for example route_entry, so make no sense to copy that to meta key, and then copy from meta key to next vector to just call actual BulkRouteCreate api without meta key, it seem convenient, but notice that in case of single non bulk api where meta_key is used, inside that api let say:  sai_status_t SaiInterface::set( _In_ const sai_object_meta_key_t& metaKey, _In_ const sai_attribute_t *attr), there i switch/case statement for object type, and actual *_entry api is called for that particular object type, but there is no data copy there back and forth, if i remember correctly thats the reason i didnt put bulk meta_key apis at the first place in SaiInterface.h class